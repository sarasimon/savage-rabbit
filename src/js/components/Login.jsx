import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-bootstrap/lib/Button';

export default class Login extends React.Component {
  constructor(props) {
    super(props);

    this.onSignInClick = this.onSignInClick.bind(this);
  }

  onSignInClick() {
    chrome.identity.getAuthToken({ interactive: true }, (token) => {
      if (chrome.runtime.lastError) {
        alert(chrome.runtime.lastError.message);
      } else {
        this.setState({ token });
        this.props.onSuccess(token);
      }
    });
  }

  render() {
    return (
      <Button bsStyle="primary" className="login" onClick={this.onSignInClick}>
        Sign in with Google
      </Button>);
  }
}

Login.propTypes = {
  onSuccess: PropTypes.func.isRequired,
};
