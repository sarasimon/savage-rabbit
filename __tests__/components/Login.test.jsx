jest.mock('chrome')
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../src/js/components/Login';


describe('Login', () => {
  test('test basic login', () => {
    const component = renderer.create(
      <Login onSuccess={() => {}} />,
    );
    /*  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    tree.props.onClick();
      
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    */
  });
});
