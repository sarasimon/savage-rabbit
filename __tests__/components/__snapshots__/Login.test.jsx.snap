// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test basic login 1`] = `
<button
  className="login btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Sign in with Google
</button>
`;
